### PR TITLE
add dynamic-linking feature

### PR DESCRIPTION
**Pull request by bbigras**
I seem to need this to build a package on nixos.

### CODE REVIEW

**Comment by houqp** [comment]
nixos fancy :D

**Comment by houqp** [comment]
could you add a note in the doc about this feature flag as well?

**Comment by bbigras** [comment]
> could you add a note in the doc about this feature flag as well?

Sure. In the `cargo doc` or https://github.com/delta-io/kafka-delta-ingest/blob/main/doc/DESIGN.md ?

**Comment by houqp** [comment]
cargo doc would be better

**Comment by bbigras** [comment]
> cargo doc would be better

Done. Feel free to suggest better wordings.

File: src/lib.rs [hunk 1]
```diff
@@ -1,4 +1,7 @@
 //! Implementations supporting the kafka-delta-ingest daemon
+//! ## Feature flags
```
Comment (houqp) [inline src/lib.rs:2]:
```suggestion

//! ## Feature flags
```

**Review by houqp — APPROVED** [review]
LGTM

**Comment by bbigras** [comment]
Any progress on the review side :pray: ?

**Comment by houqp** [comment]
Thanks @bbigras !

**Comment by bbigras** [comment]
Thanks!

**Pull request merged**